op-node/rollup/engine/payload_process.go

```diff
@@ -30,21 +30,25 @@ func (eq *EngDeriver) onPayloadProcess(ev PayloadProcessEvent) {
 		ev.Envelope.ExecutionPayload, ev.Envelope.ParentBeaconBlockRoot)
 	if err != nil {
 		eq.emitter.Emit(rollup.EngineTemporaryErrorEvent{
-			Err: fmt.Errorf("failed to insert execution payload: %w", err)})
+			Err: fmt.Errorf("failed to insert execution payload: %w", err),
+		})
 		return
 	}
 	switch status.Status {
 	case eth.ExecutionInvalid, eth.ExecutionInvalidBlockHash:
+		// TODO: Do we also want to retry deposit-only attributes at this point?
```

Comment:
OK also added a deposits-only retry here. However, I'm wondering how this can happen during derivation? Wouldn't it already fail way earlier when calling FCU? How can the FCU call that builds the payload succeed and only later this insert fail? AFAIK the FCU call during derivation is blocking (because of `noTxPool == true`) and would already return an `INVALID` status error.

Reply:
The engine error spec on `getPayload` is underspecified w.r.t. block building exceptions, I like to handle the `newPayload` errors where we can, as they're better defined.